packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -390,6 +390,10 @@ export class DeltaManager
      * @param readonly - set or clear force readonly.
      */
     public forceReadonly(readonly: boolean) {
+        this.logger.sendTelemetryEvent({
```

Comment:
can we move it down and only log if oldValue != this.readonly

Reply:
`this.readonly` is weird and we're trying to catch mismatches in readonly expectations, so I didn't want to miss calls in weird/unexpected flows.  E.g. connect as read, call `forceReadonly(true)`, then somehow reconnect as write -- want to have telemetry that reveals why we get "ops sent in readonly state" errors despite having a write connection.  I'll scope it to only when the `this._forceReadonly` value changes though just in case it's getting called repeatedly with the same value.

Reply:
read vs write on the connection is irrelevant. you care about why the deltamanger is readonly, which can be force, permission, or storage only. you probably want to log that. i would add it at the point the container is closed

Reply:
Latest adds logging in both places